text/0001-cdk-update.md

```diff
@@ -269,10 +269,27 @@ off of information in the CDK `App` context.
 ### What is the high level implementation plan?
 
 We will start from the prototype CDK update command that identifies the Lambda
-resources and then publishes them using the CDK toolkit.
-
-We will propose updates to the initial CDK constructs that are capable of
-accelerated deployment and make those changes to the CDK.
+resources and then publishes them using the CDK toolkit, and extend that to
+implement support for ECR images associated with ECS and Fargate tasks, API
+Gateway definitions, and Step Function workflows. Those will be implemented
+directly in the toolkit code as part of the launch for the feature. The toolkit
+implementation will be designed to conform to an interface that provides:
+
+- Watchable filesystem resources
+- A way of updating the watchable resource list
+- A method for determining whether the construct can be updated in place
+- A method for updating the construct in place
+
+If changes to the CDK constructs are necessary to implement the accelerated
+development process, we will make those changes as well. In the longer term we
+must lay the groundwork for moving the logic defining the update process into
+the Construct library, which implies a design for passing these values by way of
+the Cloud Assembly.
+
+We will implement a filesystem watcher for the CDK toolkit that works on one or
```

Comment:
The watch list could be language-dependent so at least we should add support for specifying a set of watch-exclusion globs that will be applied to `cdk init` templates for each language. For example, for TypeScript, we will likely want to exclude `node_modules/**/*` from watch

Reply:
Good idea.